File: UygulamaDersleri/01_Giris/Bolum2Soru10.R

# Gerekli paketler
require(MASS)
attach(Boston)
# Veri seti açiklamasi
?Boston

# Satir sayisi
nrow(Boston)
# Sütun sayisi
ncol(Boston)

# Part b
# Karsilastirma grafigi
pairs(Boston, pch = 21)
# Yüksek korelasyon katsayisina sahip olan özelliklerin grafikleri
indices <- which((abs(cor(Boston))>0.75 & abs(cor(Boston))<1), arr.ind = TRUE)
indices <- unique(indices)
pairs(Boston[,indices],pch = 21)

# Part c
sort(abs(cor(Boston)[,1]), decreasing = TRUE)
pairs(Boston[, c('rad','tax','lstat','nox','indus')])

# Part d
hist(Boston$crim, breaks = 20)
Boston[Boston$crim>40,]

hist(Boston$tax, breaks = 20)
Boston[Boston$tax>700,]

hist(Boston$ptratio, breaks = 50)
Boston[Boston$ptratio>=22,]

# Part e
nrow(Boston[Boston$chas == 1, ])

# Part f
median(Boston$ptratio)

# Part g
Boston[min(Boston$medv),]
summary(Boston)

# part h
nrow(Boston[Boston$rm>7,])
nrow(Boston[Boston$rm>8,])

Boston[Boston$rm>8,]
summary(Boston)

File: UygulamaDersleri/01_Giris/Bolum2Soru10.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jan 21 09:23:24 2020
@author: utkuk
"""
import numpy as np
import pandas as pd
from sklearn.datasets import load_boston

# Part a
boston = load_boston()
print(boston.data.shape)
print(boston.DESCR)

bos = pd.DataFrame(boston.data, columns = boston.feature_names)
print(bos.head())
bos['PRICE'] = boston.target
print(bos.head())

# Part b
pd.plotting.scatter_matrix(bos)
print(bos.describe())

indices = np.array(np.where(np.abs(bos.corr())>0.7))
indices = np.unique(indices[0,(indices[0,:] != indices[1,:])])
pd.plotting.scatter_matrix(bos.iloc[:,indices])

# Part c
corrMatris = np.abs(bos.corr().iloc[:,0])
sort = corrMatris.sort_values()
pd.plotting.scatter_matrix(bos.loc[:,['RAD', 'TAX', 'LSTAT', 'NOX', 'INDUS']])

# Part d
pd.plotting.hist_series(bos.loc[:,'CRIM'], bins = 20)
bos.loc[bos['CRIM']>40,:]

pd.plotting.hist_series(bos.loc[:,'TAX'], bins = 20)
bos.loc[bos['TAX']>700,:]

pd.plotting.hist_series(bos.loc[:,'PTRATIO'], bins = 50)
bos.loc[bos['PTRATIO']>22,:]

# Part e
sum(bos['CHAS']==1)

# Part f
bos['PTRATIO'].median()

# Part g
minPrice = np.where(bos['PRICE'] == bos['PRICE'].min())[0]
bos.iloc[minPrice,:]
print(bos.describe())

# Part h
bos.loc[bos['RM']>7,:].shape
bos.loc[bos['RM']>8,:].shape

bos.loc[bos['RM']>7,:]
print(bos.describe())